guardian/prosemirror-elements
# Lint strict boolean expressions

## What does this change?

Disallows numbers being coerced into boolean expressions. It is really easy to introduce subtle bugs by testing for non-nullness and inadvertently also  excluding `0`. Hopefully being more explicit about `null` and `0` also makes the intent of the code more clear.

`@typescript-eslint/strict-boolean-expressions` allows us to selectively disallow specific types from coercion to `boolean`. To start with, I've just excluded numbers, since these are typically the highest risk of introducing bugs. But it would be nice in future to expand this to exclude strings and objects too.

## How to test / review

Please carefully check every change, since it is easy to switch a `!variable` into a `variable !== undefined` when it should be `variable === undefined`.
